Packs/EmailCommunication/Scripts/SendEmailReply/SendEmailReply.py

```diff
@@ -222,12 +431,139 @@ def get_mailbox_from_incident_labels(labels):
     Returns:
         The mailbox label.
     """
-    for label in labels:
-        if label.get('type') == 'Mailbox':
-            return label.get('value')
+    if labels:
+        for label in labels:
+            if label.get('type') == 'Mailbox':
+                return label.get('value')
     return None
 
 
+def get_query_window():
+    """
+    Check if the user defined the list `XSOAR - Email Communication Days To Query` to give a custom value for the time
+    to query back for related incidents. If yes, use this value, else use the default value of 60 days.
+    """
+    user_defined_time = demisto.executeCommand('getList', {'listName': 'XSOAR - Email Communication Days To Query'})
+    if is_error(user_defined_time):
+        demisto.debug('Error occurred while trying to load the `XSOAR - Email Communication Days To Query` list. Using'
+                      ' the default query time - 60 days')
+        return '60 days'
+
+    try:
+        query_time = user_defined_time[0].get('Contents')
+        return f'{int(query_time)} days'
+    except ValueError:
+        demisto.error('Invalid input for number of days to query in the `XSOAR - Email Communication Days To Query` '
+                      'list. Input should be a number only, representing the number of days to query back.\nUsing the '
+                      'default query time - 60 days')
+        return '60 days'
+
+
+def get_incident_by_query(query):
+    """
+    Get a query and return all incidents details matching the given query.
+    Args:
+        query: Query for the incidents that should be returned.
+    Returns:
+        dict. The details of all incidents matching the query.
+    """
+    # In order to avoid performance issues, limit the number of days to query back for modified incidents. By default
+    # the limit is 60 days and can be modified by the user by adding a list called
+    # `XSOAR - Email Communication Days To Query` (see README for more information).
+    query_time = get_query_window()
+
+    query_from_date = str(parse_date_range(query_time)[0])
+
+    res = demisto.executeCommand("GetIncidentsByQuery", {"query": query, "fromDate": query_from_date,
+                                                         "timeField": "modified", "Contents": "id,status"})[0]
+    if is_error(res):
+        return_results(ERROR_TEMPLATE.format('GetIncidentsByQuery', res['Contents']))
+        raise DemistoException(ERROR_TEMPLATE.format('GetIncidentsByQuery', res['Contents']))
+
+    incidents_details = json.loads(res['Contents'])
+    return incidents_details
+
+
+def get_unique_code():
```

Comment:
For the new "Email Threads" layout and it's added functionality I needed to be able to generate a unique code for first-contact emails being sent from XSOAR since there would not already be one to reference.  I made copies of code from `PreprocessEmail` so that codes are being generated the same way.  I didn't see a way to do it without code re-use, other than perhaps adding these functions to `CommonServerPython`.